parachain/tests/adder.rs

```diff
@@ -200,3 +205,53 @@ fn processes_messages() {
 	assert_eq!(new_head.parent_hash, hash_head(&parent_head));
 	assert_eq!(new_head.post_state, hash_state(1024));
 }
+
+#[test]
+fn terminates_on_timeout() {
+	let result = parachain::wasm_executor::validate_candidate(
+		INFINITE_LOOP_CODE,
+		ValidationParams {
+			parent_head: Default::default(),
+			block_data: Vec::new(),
+			ingress: Vec::new(),
+		},
+		&mut DummyExt,
+		parachain::wasm_executor::ExecutionMode::RemoteTest,
+	);
+	match result {
+		Err(parachain::wasm_executor::Error::Timeout) => {},
+		_ => panic!(),
+	}
+
+	let parent_head = HeadData {
+		number: 0,
+		parent_hash: [0; 32],
+		post_state: hash_state(0),
+	};
+
+	let block_data = BlockData {
+		state: 0,
+		add: 512,
+	};
+
+	let _ret = parachain::wasm_executor::validate_candidate(
+		TEST_CODE,
+		ValidationParams {
+			parent_head: parent_head.encode(),
+			block_data: block_data.encode(),
+			ingress: Vec::new(),
+		},
+		&mut DummyExt,
+		parachain::wasm_executor::ExecutionMode::RemoteTest,
+	).unwrap();
+}
+
+// This is not an actual test, but rather an entry point for out-of process WASM executor.
+// When executing tests the executor spawns currently executing binary, which happens to be test binary.
+// It then passes "validation_worker" on CLI effectivly making rust test executor to run this single test.
+#[test]
+fn validation_worker() {
+	if let Some(id) = std::env::args().find(|a| a.starts_with("/shmem_rs_")) {
```

Comment:
Why not make the validation worker executable configurable as env variable? `VALIDATION_WORKER` and if this is set, we don't use `current_exe()`?
In our tests we would just set this to `VALIDATION_WORKER=cargo run --`.

Reply:
I'd prefer tests to not depend on the root crate. Currently they are still self-contained and don't require any additional binaries.

Reply:
Okay, in Cumulus testing I probably can just use the in-process stuff.